===== Patch instructions generated by SAVER =====
[+] { Insert: if (true) free(*(rec_int_collect_field_list:field_name)) at 3 (line 364, column 3),
        Insert: if (true) free(*(rec_int_collect_field_list:field_name)) at 15 (line 348, column 19),
        Insert: if (true) free(*(rec_int_collect_field_list:field_name)) at 21 (line 354, column 19) }
===== Actual patch in diff format =====
--- a/table1/benchmarks/recutils-1.8/src/rec-int.c
+++ b/table1/benchmarks/recutils-1.8/src/rec-int.c
@@ -347,6 +347,7 @@ rec_int_collect_field_list (rec_record_t record,
 
       for (j = 0; j < rec_fex_size (fex); j++)
         {
+          free(field_name);
           rec_fex_elem_t elem = rec_fex_get (fex, j);
           char *field_name = strdup (rec_fex_elem_field_name (elem));
 
@@ -356,11 +357,12 @@ rec_int_collect_field_list (rec_record_t record,
                                   rec_fex_elem_min (elem),
                                   rec_fex_elem_max (elem)))
             /* Not enough memory: panic and retreat!  */
+            free(field_name);
             return NULL;
         }
       rec_fex_destroy (fex);
     }
-
+  free(field_name);
   return res;
 }
 
